File: MCI Latest 1/app/src/main/java/com/greenusys/personal/registrationapp/fragments/NewsAdapter.java
package com.greenusys.personal.registrationapp.fragments;

import android.content.Context;
import android.content.Intent;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.greenusys.personal.registrationapp.FullNewzActivity;
import com.greenusys.personal.registrationapp.R;
import com.greenusys.personal.registrationapp.pojos.News;
import com.greenusys.personal.registrationapp.pojos.Newz;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by personal on 2/22/2018.
 */

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.NewsAdapterViewHolder> {

    private final Context context;

    private List<News> newsList;

    private NewItemOnClickHandler clickHandler;
    public interface NewItemOnClickHandler{

        public void onClick(Newz newz);
    }

    public NewsAdapter(Context context, NewItemOnClickHandler clickHandler,ArrayList<News> newsList)
    {
        this.context = context;
        this.newsList = newsList;

        newsList = new ArrayList<>();
        this.clickHandler = clickHandler;
    }


    @Override
    public NewsAdapterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.news_item_layout,parent,false);

        return new NewsAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(NewsAdapterViewHolder holder, int position) {



        holder.heading.setText(newsList.get(position).getTitle());
        holder.date.setText(newsList.get(position).getTime().split(" ")[0]);
        holder.description.setText(newsList.get(position).getNews());

        holder.newslayout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                Intent intent = new Intent(v.getContext(), FullNewzActivity.class);

              intent.putExtra("title",newsList.get(position).getTitle());
                intent.putExtra("date",newsList.get(position).getTime());
                intent.putExtra("description",newsList.get(position).getNews());

                v.getContext().startActivity(intent);



            }
        });




    }

    @Override
    public int getItemCount() {
        return newsList.size();
    }

    public class NewsAdapterViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

        TextView heading;
        TextView date;
        TextView description;
        ConstraintLayout newslayout;

        public NewsAdapterViewHolder(View itemView)
        {
            super(itemView);
            heading = (TextView)itemView.findViewById(R.id.headline_tv);
            date = (TextView)itemView.findViewById(R.id.date_tv);
            description = (TextView)itemView.findViewById(R.id.description_tv);
            newslayout = (ConstraintLayout) itemView.findViewById(R.id.news_layout);



           // itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View view) {

           // clickHandler.onClick(newsList.get(getAdapterPosition()));

        }
    }

    public void updateData(List<News> newsList)
    {
        this.newsList = newsList;
        notifyDataSetChanged();
    }
}

File: CurrencyConverter/app/src/main/java/com/example/currencyconverter/adapter/News_Adapter.kt
package com.example.currencyconverter.adapter

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.example.currencyconverter.R
import com.example.currencyconverter.activity.News_Description
import com.example.currencyconverter.modal.News_Resp_Modal
import kotlinx.android.synthetic.main.news_item.view.*

class News_Adapter(val list:ArrayList<News_Resp_Modal.News_Res_List>, val context:Context): RecyclerView.Adapter<News_Adapter.ViewHolder>()
{
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {

    val v=LayoutInflater.from(context).inflate(R.layout.news_item,parent,false);
    return ViewHolder(v)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {

        holder.main_layout.setOnClickListener()
        {
            context.startActivity(Intent(context,News_Description::class.java)
                    .putExtra("title",list[position].blog_title)
                    .putExtra("description",list[position].blog_content)
                    .putExtra("image","https://convertedcurrency.com/admin/"+list[position].blog_image)
            )
        }

        Glide.with(context)
                .load("https://convertedcurrency.com/admin/"+list[position].blog_image)
                .into(holder.blog_image)
        holder.blog_title.text=list[position].blog_title

    }

    class ViewHolder(view: View):RecyclerView.ViewHolder(view) {

        var main_layout:LinearLayout=view.main_layout
        var blog_image=view.blog_image
        var blog_title=view.blog_title
    }


}